richtext.md

```diff
@@ -0,0 +1,112 @@
+# Rich Text (plus refactoring and RTL/Intl support)
+
+All of our text widgets need work to clean up, make more extensible and support rich text as well as future internationalisation.
+
+## Background
+
+The goal of this work is to develop a text handling solution that will have:
+
+* Rich text support
+* Allow developers to create custom widgets incorporating the text capability
+* Good testability and performance
+* Be more maintainable than the current state of text
+
+The lead-up to this proposal has been discussed in various places
+
+* https://github.com/fyne-io/fyne/wiki/Text-Refactor
+* https://github.com/fyne-io/fyne/issues/21
+
+## Architecture / API
+
+* The BiDi/Shaping libraries
+  - built in collaboration with Gio team will underpin the drawing elements of this work, but are not described in this proposal.
+* RichText widget that encapsulates the parsing, styling and shaping for all our text requirements (see more in Design)
+* textProvider capabilities will be moved to RichText and the textPresenter interface moved to configuration calls.
+* To continue supporting widgets that expose a Text field we will support passing a \*string or \*[]rune into RichText, working around the previous need for copying and comparing large strings in Refresh.
+* The Label widget will be re-worked to simply wrap a RichText.
+
+Edit mode of rich text will be added at a later date, moving these capabilities out of the `Entry` widget and into the new `RichText`.
+
+The proposed API for `RichText` follows the familiar pattern of standard components (rendered according to theme) that can be extended as required.
+Each text segment can provide more than just text by providing a different graphical element from the `CanvasRepresentation` call.
+
+```go
+var (
+	RichTextStyleInline RichTextStyle
+	RichTextStyleParagraph RichTextStyle
+	RichTextStyleHeading RichTextStyle
+	RichTextStyleSubHeading RichTextStyle
+	RichTextStyleQuote RichTextStyle
+	RichTextStyleURL RichTextURL
+	RichTextStyleEmphasis RichTextStyle
+)
+
+type RichText struct {
+	BaseWidget
+
+	Segments []RichTextSegment
+}
+
+type RichTextSegment interface {
+	Textual() string
+	Visual() fyne.CanvasObject
+
+	Select(pos1, pos2 fyne.Position)
+	SelectedText() string
+}
+
+type TextSegment struct { // would implement RichTextSegment, handles rendering text
+	Text string
+	Type RichTextStyle
```

Comment:
Is there any reason to have `Type` instead of `Style` or `RichTextStyle` instead of `RichTextType`? Anyway, IMHO the name selected should match with the field name.

Reply:
Yes, it should probably be Style